client/engine/messageservice/test-messageservice.go

```diff
@@ -50,46 +56,30 @@ func (t TestMessageService) In() chan<- protocols.Message {
 	return t.in
 }
 
-func (t TestMessageService) Send(message protocols.Message) {
-	t.in <- message
-}
-
 // Connect creates a gochan for message service to send messages to the given peer.
-func (t TestMessageService) Connect(peer TestMessageService) {
-	toPeer := make(chan string)
-
-	t.toPeers[peer.address] = toPeer
-
+func (t TestMessageService) Connect(b Broker) {
 	go func() {
-		for msgString := range toPeer {
-			msg, _ := protocols.DeserializeMessage(msgString)
-			peer.out <- msg // send messages directly to peer's engine, using the peer's out chan
+		for message := range t.in {
+			peerChan, ok := b.services[message.To]
+			if ok {
+				msg, err := message.Serialize()
```

Comment:
Why does this serialize and then deserialize the message?

Reply:
Any true message service would have to serialize and deserialize messages!

The honest truth is that I didn't want to change behaviour, and the previous behaviour serializes messages in `forward` and then deserializes them in the anonymous goroutine within `Connect`

Reply:
That makes sense! Could we a comment that we're doing this for testing purposes?

Could we also rename some of the vars so it's a little clearer? There's `message`,`msg`,and `m` at the moment.

Reply:
> That makes sense! Could we a comment that we're doing this for testing purposes?

Definitely!

> Could we also rename some of the vars so it's a little clearer? There's `message`,`msg`,and `m` at the moment.

Sounds good

Reply:
That change was not intentional, and can easily be reverted!